src/test/groovy/io/vertx/lang/groovy/EventBusTest.groovy

```diff
@@ -66,4 +66,20 @@ class EventBusTest extends VertxTestBase {
     eventBus.send("the_address", "${val}")
     await()
   }
+
+  @Test
+  public void testComplexJson() {
+    def val = [outer: [inner: 'value'], list: ['v1', 'v2', 'v3']]
+    def eventBus = _vertx.eventBus()
+    eventBus.consumer("the_address").handler { message ->
+      def body = message.body()
+      if (body instanceof Map && body.equals(val)) {
+        testComplete()
+      } else {
+        fail()
```

Comment:
FWIW -- this doesn't actually work -- the test just hangs.  I think we'd need to do something like what's done in `AsyncTestBase` (or maybe even sub-class that here and use its completion/wait/failure routines).  Not what I'm trying to fix, but figured I'd mention it.